lib/modules/showImages.js

```diff
@@ -1818,73 +1791,45 @@ function makeMediaMovable(media, element, dragInitiater = element) {
 
 	element.classList.add('res-media-movable');
 
+	const mediaManuallyMoved = _.once(() => { media.dispatchEvent(new CustomEvent('mediaManuallyMoved', { bubbles: true })); });
+
 	addDragListener({
 		media,
 		element: dragInitiater,
 		atShiftKey: true,
-		onMove(x, y, deltaX, deltaY) { moveMedia(element, deltaX, deltaY); },
+		onMove(x, y, deltaX, deltaY) { moveMedia(element, deltaX, deltaY); mediaManuallyMoved(); },
 	});
 }
 
-function makeMediaIndependentOnResize(media, element) {
+function makeMediaIndependent(element) {
 	const wrapper = document.createElement('div');
 	const independent = document.createElement('div');
 	$(element).replaceWith(wrapper);
 	wrapper.appendChild(independent);
 	independent.appendChild(element);
 
-	const debouncedResize = frameThrottle(media.emitResizeEvent);
-
-	media.addEventListener('mediaResize', e => {
-		media.independent = true;
-
-		// $FlowIssue `mozFullScreenElement` not recognized
-		const fullscreenElement = (document.fullscreenElement || document.webkitFullscreenElement || document.mozFullScreenElement: any);
-		if (!media.offsetParent || independent.contains(fullscreenElement)) {
-			// Allowing propagation when non-visible may cause unwanted side-effects,
-			// so cancel and instead emit a new signal when expanded
-			e.stopImmediatePropagation();
-		}
-	}, true);
-
-	let lastHeight = 0;
+	independent.classList.add('res-media-independent');
+	element.style.willChange = 'transform';
+	wrapper.style.willChange = 'height';
 
-	media.addEventListener('mediaResize', () => {
-		const height = element.clientHeight;
-		if (lastHeight !== height) {
-			lastHeight = height;
-			wrapper.style.height = `${height}px`;
-		}
-
-		independent.classList.add('res-media-independent');
+	let horizontalIsVertical = false;
+	element.addEventListener('mediaRotation', (e: any) => {
+		horizontalIsVertical = e.detail;
+		updateWrapper(element.getBoundingClientRect());
 	});
 
-	const prevExpand = media.expand;
-	media.expand = () => {
-		if (media.independent) {
-			media.emitResizeEvent();
-		} else {
-			// This is a slower method to listen to resizes, as it waits till the frame after the size is set to update.
-			// Using this is however necessary when it's not possible to determine size from media events.
-			elementResizeDetector().listenTo(element, () => {
-				if (element.clientHeight !== lastHeight) media.emitResizeEvent();
-			});
-		}
-
-		window.addEventListener('resize', debouncedResize);
-
-		if (prevExpand) return prevExpand();
+	const updateWrapper = ({ width, height }) => {
+		wrapper.style.height = `${horizontalIsVertical ? width : height}px`;
+		// Force immediate reflow in order to get resizeObserver to react faster
+		window._trash_updateWrapperRefowProvocation = wrapper.getBoundingClientRect();
```

Comment:
Debugging?

Reply:
This made the whole process go a bit faster, but that's not the seemingly not case anymore.